code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -516,7 +524,7 @@ namespace <#= qualifiedNamespace #>
 
                     if (data.DirtyBit || dirtyEvents > 0)
                     {
-                        var update = new global::Improbable.Worker.Core.SchemaComponentUpdate(<#= componentDetails.ComponentId #>);
+                        var update = new global::Improbable.Worker.Core.SchemaComponentUpdate(ComponentId);
```

Comment:
Can you revert this? In-lining it avoids a function call in a potentially hot code path. 

The compiler may optimize it away, but I'm not certain

Reply:
I mean I can revert it but then we should change line 593 and 628 because its inconsistent. Also we should remove line 473 (`public override uint ComponentId => <#= componentDetails.ComponentId #>;`) where we create this variable.

Reply:
You need the variable to identify the component for use in the CustomReplicationSystem - that's really the only reason its there.

If its inconsistent then we should probably change it